### PR TITLE
Add --dry-run Functionality to TwinTrim for Safe Simulated Duplicate Removal

### PR DESCRIPTION
**Pull request by adityakalburgi**
This PR introduces a --dry-run option to TwinTrim, allowing users to simulate the duplicate removal process without making any changes. The feature is designed to give users a safe preview of what actions will be taken, ensuring they can confidently proceed with the actual duplicate removal process.

Key Changes:
Added the --dry-run flag to simulate file deletions.
Updated the core logic to handle the dry run mode.
Modified the README to include documentation 

resolves issue #58

### CODE REVIEW

**Comment by Kota-Karthik** [comment]
@adityakalburgi 
can you please add screenshots , for showing us how this new --dry-run mode looks

**Comment by adityakalburgi** [comment]
ok will share it

**Comment by adityakalburgi** [comment]
Hi! @Kota-Karthik  here is the screenshort how it works
1. when there were no duplicates in my directory
2. when i created duplicate in my directory
Screenshot 2024-10-06 192359
Screenshot 2024-10-06 192924

File: README.md [hunk 1]
```diff
@@ -9,29 +8,39 @@ TwinTrim is a powerful and efficient tool designed to find and manage duplicate
 - **Automatic or Manual Removal**: Choose to handle duplicates automatically using the `--all` flag or manually select which files to delete.
 - **Customizable Filters**: Set filters for minimum and maximum file sizes, file types, and specific filenames to exclude from the scan.
 - **Multi-Threaded Processing**: Utilizes multi-threading to quickly scan and process large numbers of files concurrently.
-- **Deadlock Prevention**: Implements locks to prevent deadlocks during multi-threaded operations, ensuring smooth and safe execution.
+- **Deadlock Prevention**: Implements locks to prevent deadlocks during multi-threaded operations, ensuring
```
Comment (techy4shri) [inline README.md:11]: recheck this line, remove the changes from here.

File: README.md [hunk 2]
```diff
@@ -9,29 +8,39 @@ TwinTrim is a powerful and efficient tool designed to find and manage duplicate
...
 ## How It Works
 
 ### Core Components
 
-1. **File Metadata Management**: 
-    - Uses `AllFileMetadata` and `FileMetadata` classes to manage file information, such as modification time and file paths.
-    - Maintains metadata in two dictionaries (`store` and `normalStore`) for handling different levels of duplicate management.
-  
-2. **File Hashing**: 
-    - Generates a unique hash for each file using MD5 to identify duplicates by content.
```
Comment (techy4shri) [inline README.md:24]: There is no change here, simply delete and paste.

File: README.md [hunk 3]
```diff
@@ -9,29 +8,39 @@ TwinTrim is a powerful and efficient tool designed to find and manage duplicate
...
+   - Maintains metadata in two dictionaries (`store` and `normalStore`) for handling different levels of duplicate management.
+
+2. **File Hashing**:
+
+   - Generates a unique hash for each file using MD5 to identify duplicates by content.
+
 3. **File Filtering**:
-    - The `FileFilter` class provides functionality to filter files based on size, type, and exclusions.
-  
+
```
Comment (techy4shri) [inline README.md:30]: again, same thing

File: README.md [hunk 4]
```diff
@@ -59,23 +68,33 @@ python twinTrim.py <directory> [OPTIONS]
...
-    ```bash
-    python twinTrim.py /path/to/directory --min-size "50kb" --max-size "500mb" --file-type "txt"
-    ```
+
+   ```bash
+   python twinTrim.py /path/to/directory --min-size "50kb" --max-size "500mb" --file-type "txt"
+   ```
+
+4. **Dry Run Simulation**:
+
```
Comment (techy4shri) [inline README.md:94]: add this under option where the optional commands are given.

**Review by techy4shri — CHANGES_REQUESTED** [review]
Make these changes and wait for further review.

**Review by adityakalburgi — COMMENTED** [review]
i have done the required changes

**Review by techy4shri — CHANGES_REQUESTED** [review]
I do not see the changes in option seciton in readme yet, kindly implement it.

**Comment by adityakalburgi** [comment]
@techy4shri DONE!! I have added it in option section..

**Review by techy4shri — APPROVED** [review]
Approved! Everything is great! Thank you @adityakalburgi

**Comment by adityakalburgi** [comment]
Hi! @Kota-Karthik can you please explain me what the issue is?..

**Comment by Kota-Karthik** [comment]
@adityakalburgi go through the changes you made and try to run the project 
You will notice , you have many errors in updated code 
you can check your code

**Pull request merged**